src/inverted_index.c

```diff
@@ -404,7 +420,7 @@ int IR_Read(void *ctx, RSIndexResult **e) {
     goto eof;
   }
   do {
-    if (BufferReader_AtEnd(&ir->br)) {
+    while (BufferReader_AtEnd(&ir->br)) {
```

Comment:
nested `while` inside `do` looks weird. Can it be moved to a different function or macro?